src/providers/clojure.rs

```diff
@@ -0,0 +1,38 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+pub struct ClojureProvider {}
+
+impl Provider for ClojureProvider {
+    fn name(&self) -> &str {
+        "Clojure"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("project.clj"))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        Ok(Some(SetupPhase::new(vec![
+            Pkg::new("leiningen"),
+            Pkg::new("jdk8"),
```

Comment:
can the jdk version be customised? is it fixed at 8?

Reply:
I've used JavaProvider as a reference, so this would be a common question about JVM providers
But worth raising!

Reply:
Ok, I managed to add support for `NIXPACKS_JDK_VERSION` environment variable & `.jdk-version` file 
Once this is merged, we need to discuss adding to JavaProvider as well 

FYI, Nix has `jdk8`, `jdk11` and just `jdk` which has the most recent version I think

Reply:
Ok, now we support all available jdk nix packages with these values
- 8
- 11
- latest